diff -ur tvheadend-4.0.8.orig/src/plumbing/transcoding.c tvheadend-4.0.8/src/plumbing/transcoding.c
--- tvheadend-4.0.8.orig/src/plumbing/transcoding.c	2015-12-16 18:33:33.000000000 +0100
+++ tvheadend-4.0.8/src/plumbing/transcoding.c	2016-04-16 19:53:13.907273675 +0200
@@ -592,10 +597,10 @@
       // Convert audio
       tvhtrace("transcode", "%04X: converting audio", shortid(t));
 
-      tvhtrace("transcode", "%04X: IN : channels=%d, layout=%" PRIi64 ", rate=%d, fmt=%d, bitrate=%d",
+      tvhtrace("transcode", "%04X: IN : channels=%d, layout=%" PRIi64 ", rate=%d, fmt=%d, bitrate=%ld",
                shortid(t), ictx->channels, ictx->channel_layout, ictx->sample_rate,
                ictx->sample_fmt, ictx->bit_rate);
-      tvhtrace("transcode", "%04X: OUT: channels=%d, layout=%" PRIi64 ", rate=%d, fmt=%d, bitrate=%d",
+      tvhtrace("transcode", "%04X: OUT: channels=%d, layout=%" PRIi64 ", rate=%d, fmt=%d, bitrate=%ld",
                shortid(t), octx->channels, octx->channel_layout, octx->sample_rate,
                octx->sample_fmt, octx->bit_rate);
 
