--- tvheadend-4.0.8/Makefile~	2018-04-29 19:28:53.000000000 +0200
+++ tvheadend-4.0.8/Makefile	2018-04-29 19:31:17.392096527 +0200
@@ -28,7 +28,7 @@
 #
 
 CFLAGS  += -g -O2 -Wunused-result
-CFLAGS  += -Wall -Werror -Wwrite-strings -Wno-deprecated-declarations
+CFLAGS  += -Wall -Wwrite-strings -Wno-deprecated-declarations
 CFLAGS  += -Wmissing-prototypes
 CFLAGS  += -fms-extensions -funsigned-char -fno-strict-aliasing
 CFLAGS  += -D_FILE_OFFSET_BITS=64
